python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,346 @@
+# This file is part of meas_extensions_shapeHSM.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+
+import galsim
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.meas.base as measBase
+import lsst.pex.config as pexConfig
+from lsst.geom import Point2I
+
+__all__ = [
+    "HsmShapeBjConfig",
+    "HsmShapeBjPlugin",
+    "HsmShapeLinearConfig",
+    "HsmShapeLinearPlugin",
+    "HsmShapeKsbConfig",
+    "HsmShapeKsbPlugin",
+    "HsmShapeRegaussConfig",
+    "HsmShapeRegaussPlugin",
+]
+
+
+class HsmShapeConfig(measBase.SingleFramePluginConfig):
+    """Base configuration for HSM shape measurement."""
+
+    shearType = pexConfig.ChoiceField[str](
+        doc="The desired method of PSF correction using GalSim. The first "
+        "three options return an e-type distortion, whereas the last option returns a g-type shear.",
```

Comment:
Similar to Arun's comment, should `shearType` be a config? What happens if you instantiate `HsmShapeBjConfig` with `shearType='LINEAR'`?  No matter what happens, I think this is confusing :)

Could `shearType` simply be a class variable instead?  (That does have the side effect of making HsmShapeConfig purely abstract, so may require an RFC unfortunately).

At the very least, I'd recommend running Config.validate() for each of the subclasses to prohibit examples like above where the class doesn't match the `shearType`.

Reply:
Done!